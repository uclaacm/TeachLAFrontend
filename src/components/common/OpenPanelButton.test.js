// import { shallow } from 'enzyme';
// import React from 'react';
// import OpenPanelButton from './OpenPanelButton';
// 
// const clickFn = jest.fn();
// 
// describe('OpenPanelButton', () => {
//   it('smoke test', () => {
//     const component = shallow(<OpenPanelButton />);
//     expect(component.exists()).toBe(true);
//   });
// 
//   it('panelOpen=true snapshot', () => {
//     const component = shallow(<OpenPanelButton panelOpen />);
// 
//     expect(component).toMatchSnapshot();
//   });
// 
//   it('panelOpen=false snapshot', () => {
//     const component = shallow(<OpenPanelButton panelOpen={false} />);
// 
//     expect(component).toMatchSnapshot();
//   });
// 
//   it('handles undefined togglePanel prop', () => {
//     const component = shallow(<OpenPanelButton panelOpen={false} />);
// 
//     const res = component.find('.editor-expand-panel-arrow');
//     expect(res).toHaveLength(1);
//     res.simulate('click');
//   });
// 
//   it('calls togglePanel on click', () => {
//     const component = shallow(<OpenPanelButton panelOpen={false} togglePanel={clickFn} />);
// 
//     const res = component.find('.editor-expand-panel-arrow');
//     expect(res).toHaveLength(1);
//     res.simulate('click');
// 
//     expect(clickFn).toHaveBeenCalled();
//   });
// });
