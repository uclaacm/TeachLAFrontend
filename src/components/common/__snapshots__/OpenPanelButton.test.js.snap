// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenPanelButton panelOpen=false snapshot 1`] = `
<div
  className="editor-expand-panel-arrow"
  title="Open Profile Panel"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [
            "navicon",
          ],
          "f0c9",
          "M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z",
        ],
        "iconName": "bars",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;

exports[`OpenPanelButton panelOpen=true snapshot 1`] = `
<div
  className="editor-expand-panel-arrow-hidden"
/>
`;
